var/spack/repos/builtin/packages/openscenegraph/package.py

```diff
@@ -59,8 +97,8 @@ def cmake_args(self):
             "-DDYNAMIC_OPENSCENEGRAPH={0}".format(shared_status),
             "-DDYNAMIC_OPENTHREADS={0}".format(shared_status),
             "-DOPENGL_PROFILE={0}".format(opengl_profile),
+            self.define_from_variant("BUILD_OSG_APPLICATIONS", "apps"),
```

Comment:
What I mean is: you added 9 variants, and this is the only Cmake option that changed. Is that correct? Is CMake auto-detecting the other dependencies? If so, can we use explicit options to enable/disable them according to the variant?